pkg/topo/options/options.go

```diff
@@ -0,0 +1,122 @@
+// SPDX-FileCopyrightText: 2020-present Open Networking Foundation <info@opennetworking.org>
+//
+// SPDX-License-Identifier: LicenseRef-ONF-Member-1.0
+
+package options
+
+import (
+	"fmt"
+
+	topoapi "github.com/onosproject/onos-api/go/onos/topo"
+)
+
+const (
+	DefaultServicePort = 5150
+	DefaultServiceHost = "onos-topo"
+)
+
+// Options topo SDK options
+type Options struct {
+	// Watch watch options
+	Watch WatchOptions
```

Comment:
Rather than sticking the options for all the different methods in a single `Options` object, we should create a different interface and options set for each method. I know it’s a lot of boilerplate, but it results in a much cleaner and safer API for users. The API as it is now allows users to set watch options on list calls and list options on watch calls. We should always make it impossible for the user to do the wrong thing if at all possible.